src/js/player.js

```diff
@@ -1623,37 +1624,31 @@ class Player extends Component {
   }
 
   /**
-   * start media playback
+   * Attempt to begin playback at the first opportunity.
    *
    * @return {Promise|undefined}
-   *         Returns a `Promise` if the browser returns one, for most browsers this will
-   *         return undefined.
+   *         Returns a `Promise` only if the browser returns one and the player
+   *         is ready to begin playback. For some browsers and all non-ready
+   *         situations, this will return `undefined`.
    */
   play() {
-    if (this.changingSrc_) {
-      this.ready(function() {
-        const retval = this.techGet_('play');
 
-        // silence errors (unhandled promise from play)
-        if (retval !== undefined && typeof retval.then === 'function') {
-          retval.then(null, (e) => {});
-        }
-      });
-
-    // Only calls the tech's play if we already have a src loaded
-    } else if (this.isReady_ && (this.src() || this.currentSrc())) {
+    // If the tech is not ready, queue up another call to `play()` for when it
```

Comment:
The key change in all this is that we kind of have a "checklist" of conditions before we can attempt playback.

First, we need the tech to be ready before anything else. When that's satisfied, `play()` gets called again.

Second, we need to make sure middleware isn't changing sources. When that's satisfied by a `loadstart`, we call `play()` again.

Finally, both conditions should be satisfied, so we can attempt playback if there's a source.